### DISCUSSION TITLE
Option to toggle showing annotations in a graph

### DISCUSSION THREAD

**Discussion by helmo** [grafana/grafana#43631] [Dashboards, Panels & Visualizations]
While annotations are very useful they can also be 'in the way' then looking at interesting parts of a graph.

I already found https://community.grafana.com/t/how-to-hide-annotations/44769 with a similar request. That was however referred to https://github.com/grafana/grafana/issues/25580 which is about the context menu for creating an annotation.

I'd like to propose an quick toggle option in the graph panel to hide annotation lines. It only has to be visible is there are annotations in the current graph.

The 'Share Panel' is another place where such an option might be useful. It would be nice if including annotations is optional when sharing a graph image.

![image](https://user-images.githubusercontent.com/104120/147978107-bd944829-4352-4809-b8d8-4199b507f5a6.png)

![image](https://user-images.githubusercontent.com/104120/147978194-9cbb44ab-ae28-4bf4-9553-53e08e703717.png)

**Comment by kustodian** [comment]
Why do you need this when there is a dashboard wide toggle for displaying/hiding annotations?

**Comment by helmo** [comment]
Thanks,

I've now found the toggle ... it's apparently hidden by default.

![image](https://user-images.githubusercontent.com/104120/149322997-8300b411-bda4-459f-a3aa-12c84b5b3068.png)


After un-checking the checkbox I get:

![image](https://user-images.githubusercontent.com/104120/149323054-a9703d88-9ec2-47b0-bc88-77b3f7a135c1.png)

**Reply by kustodian** [reply]
That checkbox should be unchecked if you want to display it. On Grafana 8.3.3 it's unchecked by default.

Anyway, I guess you can close this discussion. Right?

**Reply by helmo** [reply]
yes

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
